llpc/test/shaderdb/relocatable_shaders/TriangleVs_CheckNoteSectionForCacheHash.spvasm

```diff
@@ -0,0 +1,87 @@
+; This test case checks that amdllpc with the `-add-hash-to-elf` option correctly
+; adds a note section to ELF, which contains the cache hash and the version of LLPC.
+
+; BEGIN_SHADERTEST
+; RUN: amdllpc -spvgen-dir=%spvgendir% -gfxip=9 -unlinked -enable-relocatable-shader-elf -v -o %t.elf -add-hash-to-elf %s > %t.log && llvm-readelf -a %t.elf >> %t.log && cat %t.log | FileCheck --check-prefixes=CHECK --ignore-case %s
+
+; CHECK: .note (size = 80 bytes)
+; CHECK:     Unknown(0)                (name = llpc_cache_hash  size = 16)
+; CHECK:           0:[[hash0:[a-z0-9][a-z0-9]]][[hash1:[a-z0-9][a-z0-9]]][[hash2:[a-z0-9][a-z0-9]]][[hash3:[a-z0-9][a-z0-9]]]
+; CHECK:     Unknown(0)                (name = llpc_version  size = 8)
+; CHECK:           0:0000[[major_ver0:[a-z0-9][a-z0-9]]][[major_ver1:[a-z0-9][a-z0-9]]] 0000[[minor_ver0:[a-z0-9][a-z0-9]]][[minor_ver1:[a-z0-9][a-z0-9]]]
+
+; CHECK: Displaying notes found in: .note
+; CHECK:   Owner                Data size 	Description
+; CHECK:   llpc_cache_hash      0x00000010	Unknown note type: (0x00000000)
+; CHECK:    description data: [[hash3]] [[hash2]] [[hash1]] [[hash0]]
+; CHECK:   llpc_version         0x00000008	Unknown note type: (0x00000000)
+; CHECK:    description data: [[major_ver1]] [[major_ver0]] 00 00 [[minor_ver1]] [[minor_ver0]] 00 00
```

Comment:
I still don't think this is testing what you want.  When you use "-v" it dumps the elf section.  That is what you are checking above.  This is also dumping the contents of the elf.  So this is testing that the two ways of dumping elf give the same results.  It does not tell us if the value in the elf is the correct value.

Reply:
It is not perfect but it checks some errors. When I debug the code, many times it gives wrong note name, wrong size of note name, and different result between the dump and readelf if the code has some bugs.

Currently, I have no idea how to get the hash value directly from amdllpc or other programs. Please let me know if there is a way to do that. An alternative might be adding a new option to amdllpc and letting it print the hash value.

Reply:
The current output with "-v" has a section that output the hash for the pipeline:

```
===============================================================================
// LLPC calculated hash results (graphics pipline)

PIPE : 0x45BA1333B1857821
VS   : 0x0FCAFD9563CB7651
FS   : 0x69103766E03D91A7
```

This could be updated to output the 64-bit has as well.  Then you can have similar output when the hash is computed for relocatable shaders.

The LLPC version is not output anywhere yet, but that might be reasonable to output as well.

Reply:
Got it. Since the hash printed by `-v` is different from the hash for cache lookup, I added a few `LLPC_OUTS(..)` to correctly emit the hash for cache lookup in addition to emitting the llpc version.